src/base/utils/fs.cpp

```diff
@@ -166,10 +167,15 @@ bool Utils::Fs::sameFiles(const Path &path1, const Path &path2)
 {
     QFile f1 {path1.data()};
     QFile f2 {path2.data()};
-    if (!f1.exists() || !f2.exists()) return false;
-    if (f1.size() != f2.size()) return false;
-    if (!f1.open(QIODevice::ReadOnly)) return false;
-    if (!f2.open(QIODevice::ReadOnly)) return false;
+
+    if (!f1.exists() || !f2.exists())
+        return false;
+    if (path1 == path2) // if both paths are non-existant, result should be `false`
```

Comment:
From the function name `bool sameFiles()`, I think the point is on file existence (its contents). IMO if there are no files then the paths refers to nothing and we cannot say the files are same (because there are no files), we can only say the paths are the same.

Reply:
Ok.
But the comment in the code confuses. It is unclear what exactly is meant. Or it does not belong to this line. But it's still incomprehensible.
If you only wanted to clarify what you explained to me above, then it would be better to have it described above the function declaration (as its contract).

Reply:
Revised.